pkg/controller/common/keystore/user_secret.go

```diff
@@ -45,19 +46,30 @@ func secureSettingsVolume(
 ) (*volume.SecretVolume, string, error) {
 	// setup (or remove) watches for the user-provided secret to reconcile on any change
 	watcher := k8s.ExtractNamespacedName(hasKeystore)
-	if err := watches.WatchUserProvidedSecrets(
+
+	// user-provided Secrets referenced in the resource
+	secretSources := WatchedSecretNames(hasKeystore)
+	// user-provided Secrets referenced in a StackConfigPolicy that configures the resource
+	policySecretSources, err := filesettings.GetSecureSettingsSecretSources(ctx, r.K8sClient(), hasKeystore)
+	if err != nil {
+		return nil, "", err
```

Comment:
It might make sense to wrap this error here, as the underlying funcs called here both `json.Unmarshal`, and `strconv.ParseInt`, which are going to return some pretty generic errors which may be hard to track down if context isn't given here.